pkg/autoscaler/autoscaler/util.go

```diff
@@ -146,3 +147,54 @@ func genMetricsEndpoint(tac *v1alpha1.TidbClusterAutoScaler) (string, error) {
 	}
 	return fmt.Sprintf("http://%s-prometheus.%s.svc:9090", tac.Spec.Monitor.Name, tac.Spec.Monitor.Namespace), nil
 }
+
+func autoscalerToStrategy(tac *v1alpha1.TidbClusterAutoScaler) *pdapi.Strategy {
+	strategy := &pdapi.Strategy{
+		Resources: make([]*pdapi.Resource, 0, len(tac.Spec.Resources)),
+	}
+
+	for _, res := range tac.Spec.Resources {
+		resource := &pdapi.Resource{
+			CPU:          res.CPU.AsDec().UnscaledBig().Uint64(),
+			Memory:       res.Memory.AsDec().UnscaledBig().Uint64(),
+			Storage:      res.Storage.AsDec().UnscaledBig().Uint64(),
+			ResourceType: res.ResourceType,
+		}
+		if res.Count != nil {
+			count := uint64(*res.Count)
+			resource.Count = &count
+		}
+		strategy.Resources = append(strategy.Resources, resource)
+	}
+
+	if tac.Spec.TiDB != nil {
+		strategy.Rules = append(strategy.Rules, autoRulesToStrategyRule("tidb", tac.Spec.TiDB.Rules))
+	}
+
+	if tac.Spec.TiKV != nil {
+		strategy.Rules = append(strategy.Rules, autoRulesToStrategyRule("tikv", tac.Spec.TiKV.Rules))
+	}
+	return strategy
+}
+
+func autoRulesToStrategyRule(component string, rules map[corev1.ResourceName]v1alpha1.AutoRule) *pdapi.Rule {
+	result := &pdapi.Rule{
+		Component: component,
+	}
+	for res, rule := range rules {
+		switch res {
+		case corev1.ResourceCPU:
+			result.CPURule = &pdapi.CPURule{
+				MaxThreshold:  rule.MaxThreshold,
+				MinThreshold:  *rule.MinThreshold,
+				ResourceTypes: rule.ResourceTypes,
+			}
+		case corev1.ResourceStorage:
+			result.StorageRule = &pdapi.StorageRule{
+				MinThreshold:  1.0 - rule.MaxThreshold,
+				ResourceTypes: rule.ResourceTypes,
+			}
```

Comment:
We should directly deliver the pass `MinThreshold` to PD because user should directly define `MinThreshold`, so we can't get `MinThreshold` by calculating `1 - MaxThreshold` which also means `MaxThreshold` and `MinThreshold` should both be pointer and its existing should be verified by validating logic.

Reply:
IMO, we can keep it for now even if we don't take it into account.

Reply:
1. Agree with @rleungx to keep the storage code block, actually, there is very little extra support to support storage as it shares most of the logic with CPU.
2. For the storage, users should set MaxThreshold in the Operator API, as in this case, the MaxThreshold has the same meaning for both the CPU and storage rules, that is, when the usage exceeds the MaxThreshold, we will do auto-scaling, otherwise, we have to explain that, for CPU, we will scale when the usage exceeds the MaxThreshold and for storage, we will scale when the available space is less than the minThreshold, which is more complicated for users to understand than the current solution.

Reply:
I agree with the current design and we can keep this code